test/test_integration_credentials.py

```diff
@@ -211,30 +217,32 @@ def test_https_netrc_is_created_with_host(self):
         output = self.invoke_cli(self.cli_auth_params,
                                  ['login', '-i', 'host/somehost', '-p', f'{extract_api_key_from_message}'])
 
-        self.validate_netrc(f"{self.client_params.hostname}", "host/somehost", extract_api_key_from_message)
+        self.validate_credentials(f"{self.client_params.hostname}", "host/somehost", extract_api_key_from_message)
         self.assertIn("Successfully logged in to Conjur", output.strip())
 
+    @integration_test()
+    def test_working_with_keystore(self):
+        cred_store,_ = CredentialStoreFactory().create_credential_store()
+        self.assertEquals(type(cred_store) , type(KeystoreCredentialsProvider()))
+
     '''
-    Validates when a user can logout successfully
-    '''
+        Validates logout doesn't remove another entry not associated with Cyberark
+        '''
+
     @integration_test(True)
-    def test_https_logout_successful(self):
+    def test_https_credentials_does_not_remove_irrelevant_entry(self):
+        creds = CredentialsData(self.client_params.hostname,"admin",self.client_params.env_api_key)
+        utils.save_credentials(creds)
 
-        self.invoke_cli(self.cli_auth_params,
-                        ['login', '-i', 'admin', '-p', self.client_params.env_api_key])
-        assert os.path.exists(DEFAULT_NETRC_FILE) and os.path.getsize(DEFAULT_NETRC_FILE) != 0
+        creds = CredentialsData("somemachine", "somelogin", "somepass")
+        utils.save_credentials(creds)
 
-        output = self.invoke_cli(self.cli_auth_params,
-                                 ['logout'])
-
-        self.assertIn('Successfully logged out from Conjur', output.strip())
-        with open(DEFAULT_NETRC_FILE) as netrc_file:
-            assert netrc_file.read().strip() == "", 'netrc file is not empty!'
+        self.invoke_cli(self.cli_auth_params,
+                        ['logout'])
+        cred_store, _ = CredentialStoreFactory().create_credential_store()
+        assert cred_store.is_exists("somemachine")
+        assert not cred_store.is_exists(self.client_params.hostname)
```

Comment:
why shouldnt `self.client_params.hostname` not be in cred store? We aren't overriding anything and it should in fact exist

Reply:
if this test passes we might have a bug 🙃

Reply:
ignore this ^ we good <3